pyungo/core.py

```diff
@@ -336,16 +344,29 @@ def calculate(self, data):
         t1 = dt.datetime.utcnow()
         LOGGER.info('Starting calculation...')
         self._data = Data(data)
-        self._data.check_inputs(self.sim_inputs, self.sim_outputs)
+        self._data.check_inputs(self.sim_inputs, self.sim_outputs, self.sim_kwargs)
         if not self._sorted_dep:
             self._topological_sort()
         for items in self._sorted_dep:
             # loading node with inputs
             for item in items:
                 node = self._get_node(item)
                 inputs = [i for i in node.inputs_without_constants]
+                kwarg_values = inspect.getargspec(node._fct).defaults
+
+                if node.kwargs and kwarg_values:
+                    kwarg_names = (inspect.getargspec(node._fct)
+                                   .args[-len(kwarg_values):])
```

Comment:
Yeah I agree that there might be room for inconsistencies, and I'm a bit concerned about this as well... The reason for using this instead of `node.kwargs` is that it provides a mapping between `kwargs` and their respective default values. The order of `kwargs` from `node.kwargs` may be different, and the order matters. `inspect.signature()` from [python 3](https://docs.python.org/3/library/inspect.html#inspect.signature) comes in very handy, but python 2 doesn't have it yet :disappointed:

Reply:
Your PR actually made me think that we should probably remove Python2 support in future releases 😄

Reply:
^^ I'd love that :wink: